### PR TITLE
bugfix: propagate eds changes when annotations change.

### PR DESCRIPTION
**Pull request by yuval-k**
As the lb hash propagates to EndpointsVersion, it needs to include cluster name as well, until we hopfully stop appending upstream hashes to cluster names soon

BOT NOTES: 
resolves https://github.com/solo-io/solo-projects/issues/7301

### CODE REVIEW

File: projects/gateway2/krtcollections/endpoints.go [hunk 1]
```diff
@@ -136,12 +136,14 @@ type EndpointsForUpstream struct {
 func NewEndpointsForUpstream(us UpstreamWrapper, logger *zap.Logger) *EndpointsForUpstream {
 	// start with a hash of the cluster name. technically we dont need it for krt, as we can compare the upstream name. but it helps later
 	// to compute the hash we present envoy with.
+	// add the upstream hash to the clustername, so that if it changes the envoy cluster will become warm again.
+	clusterName := GetEndpointClusterName(us.Inner)
+
 	h := fnv.New64()
 	h.Write([]byte(us.Inner.GetMetadata().Ref().String()))
+	h.Write([]byte(clusterName))
```
Comment (sam-heilbron) [inline projects/gateway2/krtcollections/endpoints.go:144]: As I see it written today (let me know if I'm misunderstanding something), we have the following data in our hash:
- The upstream name/namespace string (us-namespace_us-name)
- The cluster name which includes:
  - A hash of the upstream object
  - the cluster name for the upstream (which is just taking the us-namespace and us-name and joining them in as string)
 
Given that, it feels like we are duplicating information in the hash unnecessarily and I fear might lead to confusion for developers. What is the information that must go into this has to make it unique?

Right now we have 2 methods in different packages with the same name that do different things. I think at a minimum we should add comments to the functions to explain when they should be used (and not used)

**Review by sam-heilbron — COMMENTED** [review]
Do you think it's worth it to add an e2e test to ensure this doesn't regress in the future since it's a core piece of functionality?

File: projects/gateway2/krtcollections/endpoints.go (hunk 1 above)
Comment (yuval-k) [reply projects/gateway2/krtcollections/endpoints.go:144]: we can review tomorrow on a huddle, but the gist of it that the need ot add the cluster name is a bit of hack due to an envoy bug... let's discuss and see if there's a better way here

**Review by sam-heilbron — APPROVED** [review]

File: projects/gateway2/krtcollections/endpoints.go (hunk 1 above)
Comment (sam-heilbron) [reply projects/gateway2/krtcollections/endpoints.go:147]: We spoke offline and cleared this up. the implementation is good with me

**Comment by sam-heilbron** [comment]
> Do you think it's worth it to add an e2e test to ensure this doesn't regress in the future since it's a core piece of functionality?

We agreed that an e2e test would be valuable. However, this PR fixes the bug and disables a very flakey test. So our plan is:
- Yuval will merge this code as is, without an e2e test
- Yuval will work on fixing the flaky test
- Sam/Yuval will pair on writing an e2e test

**Review by davidjumani — APPROVED** [review]

**Pull request merged**